rule/value.go

```diff
@@ -69,35 +69,49 @@ func ParseGlobExpr(e bzl.Expr) (GlobValue, bool) {
 		return GlobValue{}, false
 	}
 	var glob GlobValue
+	parseStringsList := func(list *bzl.ListExpr) []string {
+		parsed := make([]string, 0, len(list.List))
+		for _, e := range list.List {
+			if str, ok := e.(*bzl.StringExpr); ok {
+				parsed = append(parsed, str.Value)
+			}
+		}
+		return parsed
+	}
+
+	// Positional arguments needs to be placed before named arguments, otherwise these are ambigious
+	allowPositionalArgs := true
 	for i, arg := range call.List {
-		list, ok := arg.(*bzl.ListExpr)
-		if i == 0 && ok {
-			glob.Patterns = make([]string, 0, len(list.List))
-			for _, e := range list.List {
-				if str, ok := e.(*bzl.StringExpr); ok {
-					glob.Patterns = append(glob.Patterns, str.Value)
-				}
+		if list, ok := arg.(*bzl.ListExpr); ok && allowPositionalArgs {
+			switch i {
+			case 0:
+				glob.Patterns = parseStringsList(list)
+			case 1:
+				glob.Excludes = parseStringsList(list)
+				// Last handled positional argument, no need to visit more
+				return glob, true
 			}
 			continue
 		}
+		allowPositionalArgs = false
```

Comment:
I think this should move down to after we've matched the first `AssignExpr`. We might reach this line after seeing a positional argument that's not a `ListExpr`, for example:

```
include = ["*.txt"]
exclude = ["*.c"]
glob(include, exclude)
```

Reply:
Good catch. We won't be able to read what's under the identifiers, but now we can still get content of the second positional argument. Added test case for that